showcase/src/app/shared/example-viewer/example-viewer.component.spec.ts

```diff
@@ -1,80 +1,147 @@
+import { throwError, of, Observable, Subject } from "rxjs";
+import { filter, delay } from "rxjs/operators";
+import { STARK_LOGGING_SERVICE, StarkLoggingService } from "@nationalbankbelgium/stark-core";
+import { MockStarkLoggingService } from "@nationalbankbelgium/stark-core/testing";
+import { StarkPrettyPrintModule } from "@nationalbankbelgium/stark-ui";
 
-import { ExampleViewerComponent } from "./example-viewer.component";
+import { ExampleFile, ExampleViewerComponent } from "./example-viewer.component";
 import { FileService } from "./file.service";
-import { throwError, of } from "rxjs";
-import { MockStarkLoggingService } from "@nationalbankbelgium/stark-core/testing";
-import { STARK_LOGGING_SERVICE, StarkLoggingService } from "@nationalbankbelgium/stark-core";
 import SpyObj = jasmine.SpyObj;
-import { StarkPrettyPrintModule } from "@nationalbankbelgium/stark-ui";
-import { NO_ERRORS_SCHEMA } from "@angular/core";
+import Spy = jasmine.Spy;
 
 describe("ExampleViewerComponent", () => {
 	let component: ExampleViewerComponent;
-	let fileService: FileService;
+	let fileService: SpyObj<FileService>;
 	let fixture: ComponentFixture<ExampleViewerComponent>;
 	let logger: SpyObj<StarkLoggingService>;
 
 	beforeEach(async(() => {
 		return TestBed.configureTestingModule({
 			declarations: [ExampleViewerComponent],
-			imports: [BrowserAnimationsModule, HttpClientModule, MatButtonModule, MatTabsModule, MatTooltipModule, StarkPrettyPrintModule],
+			imports: [NoopAnimationsModule, MatButtonModule, MatTabsModule, MatTooltipModule, StarkPrettyPrintModule],
 			providers: [
-				{ provide: ComponentFixtureAutoDetect, useValue: true },
 				{ provide: STARK_LOGGING_SERVICE, useValue: new MockStarkLoggingService() },
-				FileService
+				{
+					provide: FileService,
+					useValue: jasmine.createSpyObj("FileServiceSpy", ["fetchFile"])
+				}
 			],
 			schemas: [NO_ERRORS_SCHEMA] // tells the Angular compiler to ignore unrecognized elements and attributes: mat-icon
 		}).compileComponents();
 	}));
 
 	beforeEach(() => {
 		logger = TestBed.get(STARK_LOGGING_SERVICE);
+		fileService = TestBed.get(FileService);
+		fileService.fetchFile.and.callFake(() => of("initial dummy file content"));
+
 		fixture = TestBed.createComponent(ExampleViewerComponent);
 		component = fixture.componentInstance;
-		fileService = fixture.debugElement.injector.get<FileService>(FileService as any);
+		fixture.detectChanges(); // trigger initial data binding
 	});
 
-	describe("@Input() title", () => {
-		it("should change the title according to the @Input", () => {
-			const h2: any = fixture.nativeElement.querySelector(".example-viewer-title-spacer h2");
-			component.title = "Test title";
+	describe("@Input() exampleTitle", () => {
+		it("should change the exampleTitle according to the @Input", () => {
+			const h2: HTMLHeadingElement = fixture.nativeElement.querySelector(".example-viewer-title-spacer h2");
+			component.exampleTitle = "Test exampleTitle";
```

Comment:
`"Test exampleTitle"` does not really make sense. Could you please put back `"Test title"` which has more sense ? 😊

Reply:
Ah true, the "find-replace" did a bit too much :p  I'll fix it